spec/models/book_spec.rb

```diff
@@ -0,0 +1,56 @@
+require 'rails_helper'
+
+describe Book do
+  subject(:book) { build(:book) }
+  subject(:invalid_year_book) { build(:book, year: '125.4') }
```

Comment:
i think that instead doing all of this cases you could read this https://github.com/thoughtbot/shoulda-matchers and build the validations like that link explains it's easier for each validation